### PR TITLE
013: Degree PROGRAM Schema and Basic API (Backend)

### PR DESCRIPTION
**Pull request by DaleCS**
User story description:
As a developer, I need to have a schema for 'degree program' that contains related information about degree programs as well as basic API such as creating, fetching, editting, delete, etc. so that I can build complex functionalities using that schema.

### CODE REVIEW

**Comment by DaleCS** [comment]
## What I did:
* Created the schema for the degree program.
* Created corresponding API for degree program (Create, fetch, edit, delete)
     * Including validation and sanitation of inputs

### Some miscellaneous changes:
* I added a .catch statement when connecting to the database to catch an exception that occurs when your server fails to connect to the database.
* I added some scripts to our package.json(s) to allow my windows machine to run the server using terminal commands that works for windows only.

File: server/controller/program.controller.js [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+
+const programController = express.Router();
+
+/**
+ * @route   POST /program/create/
+ * @desc    Creates a new degree program
+ * @access  Private
+ */
+programController.post(
+  '/create',
```
Comment (calvinqc) [inline server/controller/program.controller.js]: usually, the standard for POST to create an object doesn't need to have `/create`, we can just use `'/'` -> Developers will know it's POST/ for create an object.

File: server/controller/program.controller.js [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+  }
+);
+
+/**
+ * @route   DELETE /program/delete/
+ * @desc    Deletes degree programs identified by it's id
+ * @access  Private
+ */
+programController.delete(
+  '/delete',
```
Comment (calvinqc) [inline server/controller/program.controller.js]: Same as above, you can just use `'/'` for delete the program

**Review by calvinqc — COMMENTED** [review]
Looks Good To Me (LGTM). 

Just a few comments that you can check out, and decide to change it or not. 

If you change it, Feel free to re-push to your branch, and `Squash and Merge` to master.

**Review by calvinqc — APPROVED** [review]

**Pull request merged**